File: src/backend/src/main/java/br/com/grupopipa/financeiro/dto/person/PersonGridDTO.java
package br.com.grupopipa.financeiro.dto.person;

import br.com.grupopipa.financeiro.dto.DTO;
import br.com.grupopipa.financeiro.entity.PersonEntity;
import br.com.grupopipa.financeiro.enumeration.PersonTypeEnum;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.UUID;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class PersonGridDTO implements DTO<PersonEntity> {

    private UUID id;
    private String name;
    private String fantasyName;
    private PersonTypeEnum personType;

    @Override
    public UUID getId() {
        return this.id;
    }

    @Override
    public void fillFromEntity(PersonEntity entity) {
        this.setId(entity.getId());
        this.setFantasyName(entity.getFantasyName());
        this.setPersonType(entity.getPersonType());
        this.setName(entity.getName());
    }
}

File: src/frontend/lib/model/person/person_grid_dto.dart
import 'package:frontend/enumeration/person_type_enum.dart';
import 'package:frontend/model/base_grid_dto.dart';
import 'package:frontend/model/person/person_dto.dart';

class PersonGridDTO implements BaseGridDTO {
  String? id;
  String? tradeName;
  String? name;
  PersonTypeEnum? personType;

  PersonGridDTO();

  PersonGridDTO.fromDTO(PersonDTO dto) {
    id = dto.id;
    tradeName = dto.tradeName;
    personType = dto.personType;
    name = dto.name;
  }

  @override
  void fillFromJson(Map<String, dynamic> map) {
    id = map["id"] as String;
    if (map["tradeName"] != null) {
      tradeName = map["tradeName"] as String;
    }
    name = map["name"] as String;
    personType = PersonTypeEnum.fromString(map["personType"] as String);
  }

  @override
  String getId() {
    return id!;
  }
}
